### ISSUE TITLE
Rancher cannot add EKS cluster in proxied environment

### ISSUE THREAD

**Issue by nickngch** [rancher/rancher#23293]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
1. have a proxied environment where outbound communication passes through a proxy
2. have proxy use a private CA
3. Try to add EKS cluster on Rancher UI

**Result:**
Error appeared on UI
`Error creating config map: Post https://xxx.yyy.eks.amazonaws.com/api/v1/namespaces/kube-system/configmaps: x509: certificate signed by unknown authority
`

**Other details that may be helpful:**
Tried:
1. import EKS CA into Rancher and Rancher agent pods.
2. import EKS CA cert into proxy VM
3. import ca-additional.pem into Rancher agent pods

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.2.8
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Hosted

gz#6166

**Comment by soumyalj** [comment]
Reproduced the issue in Rancher v2.2.8. 
1. Created Proxy server without Certs 
2. Created rancher server behind the proxy server. 
3. Created EKS cluster . Cluster creation failed with logs as below:
```
2019/12/17 21:04:33 [ERROR] ClusterController c-mzfvh [user-controllers-controller] failed with : failed to start user controllers for cluster c-mzfvh: failed to contact server: Get https://F15B9900D2BA0662566B3E7B636730E3.yl4.us-west-2.eks.amazonaws.com/api/v1/namespaces/kube-system?timeout=30s: proxyconnect tcp: waiting for cluster agent to connect
2019/12/17 21:04:33 [INFO] Stopping cluster agent for c-mzfvh
2019/12/17 21:04:33 [ERROR] ClusterController c-mzfvh [user-controllers-controller] failed with : failed to start user controllers for cluster c-mzfvh: failed to contact server: Get https://F15B9900D2BA0662566B3E7B636730E3.yl4.us-west-2.eks.amazonaws.com/api/v1/namespaces/kube-system?timeout=30s: proxyconnect tcp: waiting for cluster agent to connect
2019/12/17 21:04:38 [INFO] kontainerdriver amazonelasticcontainerservice listening on address 127.0.0.1:10467
2019/12/17 21:04:38 [INFO] kontainerdriver amazonelasticcontainerservice stopped
2019/12/17 21:04:40 [INFO] Stopping cluster agent for c-mzfvh
2019/12/17 21:04:40 [ERROR] ClusterController c-mzfvh [user-controllers-controller] failed with : failed to start user controllers for cluster c-mzfvh: failed to contact server: Get https://F15B9900D2BA0662566B3E7B636730E3.yl4.us-west-2.eks.amazonaws.com/api/v1/namespaces/kube-system?timeout=30s: proxyconnect tcp: waiting for cluster agent to connect
```
Upgraded Rancher server to v2.3-head. After upgrading the EKS cluster was active and functional.

**Comment by soumyalj** [comment]
Reproduced the issue in Rancher v2.3.2 using a HA setup.
1. Created Proxy server using private CA. (Used squid)
2. Created rancher server behind the proxy server.
```
helm install rancher rancher-stable/rancher --namespace cattle-system --set hostname=abc.net --set proxy="http://proxyserver:3128/" --set noProxy="127.0.0.0/8" --set additionalTrustedCAs=true --version 2.3.2

```
3. Created EKS cluster . Cluster creation failed as below:
```
soumyas-MBP:rke soumya$ kubectl logs rancher-675865fd58-hmtsq -n cattle-system --tail 10 -f
2020/01/02 04:28:12 [INFO] Waiting for cluster to finish provisioning
2020/01/02 04:28:12 [INFO] Cluster provisioned successfully
2020/01/02 04:28:12 [INFO] Creating worker nodes
2020/01/02 04:28:17 [INFO] Applying ConfigMap
2020/01/02 04:28:17 [INFO] kontainerdriver amazonelasticcontainerservice stopped
2020/01/02 04:28:17 [ERROR] ClusterController c-8d8zk [cluster-provisioner-controller] failed with : error creating config map: Post https://4778B80A88B44667AE5CF8517E8A0C78.yl4.us-west-2.eks.amazonaws.com/api/v1/namespaces/kube-system/configmaps: x509: certificate signed by unknown authority
2020/01/02 04:28:22 [INFO] kontainerdriver amazonelasticcontainerservice listening on address 127.0.0.1:46345
2020/01/02 04:28:22 [INFO] kontainerdriver amazonelasticcontainerservice stopped
2020/01/02 04:28:27 [INFO] kontainerdriver amazonelasticcontainerservice listening on address 127.0.0.1:41105
2020/01/02 04:28:27 [INFO] kontainerdriver amazonelasticcontainerservice stopped

```
Upgraded Rancher server to v2.3.4-rc5. After upgrading, EKS cluster did not become active. It was still stuck in provisioning state 
Rancher server logs:
```
2020/01/02 05:12:14 [INFO] Creating worker nodes
2020/01/02 05:12:19 [INFO] Applying ConfigMap
2020/01/02 05:12:19 [INFO] kontainerdriver amazonelasticcontainerservice stopped
2020/01/02 05:12:19 [ERROR] ClusterController c-8d8zk [cluster-provisioner-controller] failed with : error creating config map: Post https://4778B80A88B44667AE5CF8517E8A0C78.yl4.us-west-2.eks.amazonaws.com/api/v1/namespaces/kube-system/configmaps: x509: certificate signed by unknown authority
2020/01/02 05:12:24 [INFO] kontainerdriver amazonelasticcontainerservice listening on address 127.0.0.1:41871
2020/01/02 05:12:24 [INFO] kontainerdriver amazonelasticcontainerservice stopped
2020/01/02 05:12:29 [INFO] kontainerdriver amazonelasticcontainerservice listening on address 127.0.0.1:35193
2020/01/02 05:12:29 [INFO] kontainerdriver amazonelasticcontainerservice stopped

```
![Screen Shot 2020-01-01 at 9 20 07 PM](https://user-images.githubusercontent.com/18536626/71652748-868c7280-2cdc-11ea-89dc-beab15772972.png)

**Comment by deniseschannon** [comment]
We will not fix the upgrade use case where you have a broken cluster and upgrading Rancher would fix it. 

But can you provision an eks cluster in v2.3.4-rc5?

**Comment by soumyalj** [comment]
EKS cluster creation fails in v2.3.4-rc5. The rancher server is behind a proxy server using private CA

```
2020/01/02 18:52:42 [INFO] kontainerdriver amazonelasticcontainerservice stopped
2020/01/02 18:52:42 [ERROR] ClusterController c-fv2nr [cluster-provisioner-controller] failed with : error creating config map: Post https://9E1DBA4E3A179F915ED5FB1A5A6763B7.gr7.us-west-2.eks.amazonaws.com/api/v1/namespaces/kube-system/configmaps: x509: certificate signed by unknown authority
2020/01/02 18:52:47 [INFO] kontainerdriver amazonelasticcontainerservice listening on address 127.0.0.1:39811
2020/01/02 18:52:47 [INFO] kontainerdriver amazonelasticcontainerservice stopped
2020/01/02 18:53:03 [INFO] kontainerdriver amazonelasticcontainerservice listening on address 127.0.0.1:42379
2020/01/02 18:53:03 [INFO] kontainerdriver amazonelasticcontainerservice stopped
2020/01/02 18:53:52 [INFO] kontainerdriver amazonelasticcontainerservice listening on address 127.0.0.1:43193
2020/01/02 18:53:52 [INFO] kontainerdriver amazonelasticcontainerservice stopped
2020/01/02 18:53:52 [INFO] Provisioning cluster [c-fv2nr]
2020/01/02 18:53:52 [INFO] Creating cluster [c-fv2nr]
2020/01/02 18:53:57 [INFO] kontainerdriver amazonelasticcontainerservice listening on address 127.0.0.1:37213
2020/01/02 18:53:57 [ERROR] Cluster c-fv2nr previously failed to create
2020/01/02 18:53:57 [INFO] Starting create
2020/01/02 18:53:57 [INFO] Bringing up vpc
2020/01/02 18:54:02 [INFO] Creating service role
2020/01/02 18:54:08 [INFO] Creating EKS cluster
2020/01/02 18:54:38 [INFO] Waiting for cluster to finish provisioning
2020/01/02 18:54:38 [INFO] Cluster provisioned successfully
2020/01/02 18:54:38 [INFO] Creating worker nodes
2020/01/02 18:54:43 [INFO] Applying ConfigMap
2020/01/02 18:54:43 [INFO] kontainerdriver amazonelasticcontainerservice stopped
2020/01/02 18:54:43 [ERROR] ClusterController c-fv2nr [cluster-provisioner-controller] failed with : error creating config map: Post https://9E1DBA4E3A179F915ED5FB1A5A6763B7.gr7.us-west-2.eks.amazonaws.com/api/v1/namespaces/kube-system/configmaps: x509: certificate signed by unknown authority
2020/01/02 18:54:48 [INFO] kontainerdriver amazonelasticcontainerservice listening on address 127.0.0.1:45143
2020/01/02 18:54:48 [INFO] kontainerdriver amazonelasticcontainerservice stopped
2020/01/02 18:54:53 [INFO] kontainerdriver amazonelasticcontainerservice listening on address 127.0.0.1:34327
2020/01/02 18:54:53 [INFO] kontainerdriver amazonelasticcontainerservice stopped

```
<img width="1821" alt="Screen Shot 2020-01-02 at 10 56 07 AM" src="https://user-images.githubusercontent.com/18536626/71685924-86c55600-2d4e-11ea-9d77-a1e3f32edc2c.png">

**Comment by maggieliu** [comment]
@sowmyav27 can you verify this is working now? I believe we fixed it in an earlier 2.4.x release.

**Comment by sowmyav27** [comment]
**On 2.4-head - commit id: `b6f658307` `v2.4-4278-head` **

- Deploy a airgapped Rancher setup
- The Rancher Server URL points to an internet facing URL.
- Deploy an EKS cluster. Cluster comes up Active. 

2.5 and Master was validated here - https://github.com/rancher/rancher/issues/29070
